Content.Client/Parallax/Managers/ParallaxManager.cs

```diff
@@ -15,91 +18,74 @@
 {
-    internal sealed class ParallaxManager : IParallaxManager
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IConfigurationManager _configurationManager = default!;
+
+    private string _parallaxName = "";
+    public string ParallaxName
     {
-        [Dependency] private readonly IResourceCache _resourceCache = default!;
-        [Dependency] private readonly ILogManager _logManager = default!;
-        [Dependency] private readonly IConfigurationManager _configurationManager = default!;
+        get => _parallaxName;
+        set
+        {
+            LoadParallaxByName(value);
+        }
+    }
 
-        private static readonly ResourcePath ParallaxConfigPath = new("/parallax_config.toml");
+    public Vector2 ParallaxAnchor { get; set; }
 
-        // Both of these below are in the user directory.
-        private static readonly ResourcePath ParallaxCachedImagePath = new("/parallax_cache.png");
-        private static readonly ResourcePath PreviousParallaxConfigPath = new("/parallax_config_old");
+    private ParallaxLayerPrepared[] _parallaxLayersHQ = {};
+    private ParallaxLayerPrepared[] _parallaxLayersLQ = {};
 
-        public event Action<Texture>? OnTextureLoaded;
-        public Texture? ParallaxTexture { get; private set; }
+    public ParallaxLayerPrepared[] ParallaxLayers => _configurationManager.GetCVar(CCVars.ParallaxLowQuality) ? _parallaxLayersLQ : _parallaxLayersHQ;
 
-        public async void LoadParallax()
-        {
-            if (!_configurationManager.GetCVar(CCVars.ParallaxEnabled))
-                return;
-
-            var parallaxConfig = GetParallaxConfig();
-            if (parallaxConfig == null)
-                return;
+    public async void LoadParallax()
+    {
+        await LoadParallaxByName("default");
+    }
 
-            var debugParallax = _configurationManager.GetCVar(CCVars.ParallaxDebug);
+    private async Task LoadParallaxByName(string name)
+    {
+        _parallaxName = name;
+        Logger.InfoS("parallax", $"Loading parallax {name}");
 
-            if (debugParallax
-                || !_resourceCache.UserData.TryReadAllText(PreviousParallaxConfigPath, out var previousParallaxConfig)
-                || previousParallaxConfig != parallaxConfig)
-            {
-                var table = Toml.ReadString(parallaxConfig);
-                await UpdateCachedTexture(table, debugParallax);
+        var parallaxPrototype = _prototypeManager.Index<ParallaxPrototype>(name);
 
-                //Update the previous config
-                using var writer = _resourceCache.UserData.OpenWriteText(PreviousParallaxConfigPath);
-                writer.Write(parallaxConfig);
-            }
+        var hq = await LoadParallaxLayers(parallaxPrototype.Layers.ToArray());
+        var lq = parallaxPrototype.LayersLQUseHQ ? hq : await LoadParallaxLayers(parallaxPrototype.LayersLQ.ToArray());
```

Comment:
Just change the method to take in `List<T>`. No need to `.ToArray()` here.

Reply:
Oh, right, this was done